azurelinuxagent/ga/monitor.py

```diff
@@ -206,18 +206,20 @@ def init_sysinfo(self):
             self.sysinfo.append(TelemetryEventParam('ImageOrigin',
                                                     vminfo.image_origin))
         except (HttpError, ValueError) as e:
-            logger.warn("failed to get IMDS info: {0}", e)
+            logger.warn("failed to get IMDS info: {0}", ustr(e))
 
-    def collect_event(self, evt_file_name):
+    @staticmethod
+    def collect_event(evt_file_name):
         try:
-            logger.verbose("Found event file: {0}", evt_file_name)
+            logger.verbose("Found event file: {0}".format(evt_file_name))
             with open(evt_file_name, "rb") as evt_file:
                 # if fail to open or delete the file, throw exception
                 data_str = evt_file.read().decode("utf-8", 'ignore')
-            logger.verbose("Processed event file: {0}", evt_file_name)
+            logger.verbose("Processed event file: {0}".format(evt_file_name))
             os.remove(evt_file_name)
             return data_str
         except IOError as e:
+            os.remove(evt_file_name)
```

Comment:
why remove only on IOError? will decode() raise this exception?

Reply:
We never get any error as we use `ignore` param in decode - `decode("utf-8", 'ignore')`.

I have still added another exception `UnicodeDecodeError` as a precaution. We can decide to remove the `ignore` flag.

Changing the `except IOError` to `except (IOError, UnicodeDecodeError)`